sormas-ui/src/main/java/de/symeda/sormas/ui/person/PersonEditForm.java

```diff
@@ -631,6 +648,95 @@ protected void addFields() {
+					ApproximateAgeHelper.getApproximateAge(birthDate, (Date) getFieldGroup().getField(PersonDto.DEATH_DATE).getValue());
+			if ((pair.getElement0() != null) && (pair.getElement1() == ApproximateAgeType.YEARS)) {
+				return pair.getElement0();
+			}
+		}
+		return -1;
+	}
+
+	private void onEmancipatedChange() {
+		boolean isEmancipatedHidden = (isEmancipated == null) || (!isEmancipated.getValue());
+		boolean isEmancipatedChecked = (isEmancipated != null) && (isEmancipated.getValue());
+		hasGuardian.setValue(isEmancipatedHidden);
+		if(isEmancipatedHidden || isEmancipatedChecked) {
+			nameOfGuardians.setValue("");
+		}
+		updateHasGuardianCheckBox();
+	}
+
+	private void updateIsIncapacitatedCheckBox(boolean isInitialized) {
+		boolean isIncapacitatedChecked = (isIncapacitated == null) || (isIncapacitated.getValue());
+		int approximateAge = getApproximateAgeInYears();
+		boolean canBeEmancipated;
+		if(approximateAge == -1) {
+			canBeEmancipated = false;
+		} else {
+			canBeEmancipated = approximateAge >= minimumEmancipatedAge && approximateAge < minimumAdultAge;
+		}
+		isEmancipated.setVisible(!isIncapacitatedChecked && canBeEmancipated);
+		hasGuardian.setValue(isIncapacitatedChecked);
+		if(getApproximateAgeInYears() < minimumAdultAge) {
+			nameOfGuardians.setValue(getValue().getNamesOfGuardians());
+		}
+		updateHasGuardianCheckBox();
+		hardResetNameOfGuardians(isInitialized);
+	}
+
+	private void hardResetNameOfGuardians(boolean isInitialized) {
+		boolean isIncapacitatedChecked = (isIncapacitated != null) && (isIncapacitated.getValue());
+		if (getApproximateAgeInYears() != -1 && getApproximateAgeInYears() >= minimumAdultAge && !isIncapacitatedChecked && isInitialized) {
+			nameOfGuardians.setValue("");
+		}
+	}
+
+	private void updateHasGuardianCheckBox() {
+		boolean isIncapacitatedChecked = (isIncapacitated != null) && (isIncapacitated.getValue());
+		boolean isEmancipatedChecked = (isEmancipated != null) && (isEmancipated.getValue());
+		if((getApproximateAgeInYears() >= minimumAdultAge || (getApproximateAgeInYears() < minimumEmancipatedAge) && isEmancipatedChecked)) {
+			isEmancipatedChecked = false;
+			isEmancipated.setValue(Boolean.FALSE);
+			isIncapacitated.setVisible(true);
+		}
+		if(isEmancipatedChecked){
+			hasGuardian.setVisible(false);
+			nameOfGuardians.setVisible(false);
+			isIncapacitated.setVisible(false);
+			return;
+		} else {
+			isIncapacitated.setVisible(true);
+		}
+		Date birthDate = calcBirthDateValue();
+		boolean isChildOrUnknownDate = getApproximateAgeInYears() < minimumAdultAge;
+		nameOfGuardians.setVisible(isChildOrUnknownDate || isIncapacitatedChecked);
+		hasGuardian.setVisible(isChildOrUnknownDate || isIncapacitatedChecked);
+		if((birthDate == null) || isChildOrUnknownDate ){
+			boolean isEmancipatedVisible = ((isEmancipated != null) && (isEmancipated.isVisible()));
+			if(isEmancipatedVisible) {
+				hasGuardian.setValue(isChildOrUnknownDate);
```

Comment:
please review this listener as you update the value of the field whose change triggers this method. In other words you update the legalGuardian field in a method triggered by a change of the legalGuardian this could easily result in an infinite loop.